src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUser_WithLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidLoginLength_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithPasswordNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword(null);
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidPasswordLength_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("qwert");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAgeNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAge_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllInvalidFields_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("qwert");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAgeHigherThanIntegerMaxValue_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(Integer.MAX_VALUE + 10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
```

Comment:
This test case `registerUser_WithInvalidAgeHigherThanIntegerMaxValue_NotOk` is not valid because `Integer.MAX_VALUE + 10` will cause an integer overflow and result in an incorrect value. Java does not throw an exception for integer overflow by default.